### PR TITLE
Replace usage of `CurrentUserStore` with `useCurrentUser` hook and remove no longer needed mocks.

### PR DESCRIPTION
**Pull request by linuspahl**
## Motivation and Context

With this PR we are mainly removing no longer needed `CurrentUserStore` mocks.

There are multiple reasons why we no longer need these mocks.
- We removed the `DateTime` call which implemented the `CurrentUserStore`.
- Instead of consuming the store directly we are now using a hook which provides the current user.
- Before implementing the hook we were consuming the `CurrentUserContext`.

Please note, there are a few more files which use the `CurrentUserStore`. It requires a bigger refactoring to implement the `useCurrentUser` hook for these as well. I will continue the refactoring in a follow-up PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: graylog2-web-interface/src/contexts/CurrentUserPreferencesProvider.test.tsx [hunk 1]
```diff
@@ -67,30 +72,25 @@ describe('CurrentUserPreferencesProvider', () => {
   });
 
   it('provides default user preferences if the user has none', () => {
-    asMock(CurrentUserStore.getInitialState).mockReturnValue({ currentUser: {} as UserJSON });
+    asMock(useCurrentUser).mockReturnValue({} as User);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/contexts/CurrentUserPreferencesProvider.test.tsx:75]: What about providing a valid user, but with undefined preferences instead?

File: graylog2-web-interface/src/pages/StreamEditPage.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import { Alert } from 'components/bootstrap';
+import StreamRulesEditor from 'components/streamrules/StreamRulesEditor';
+import { DocumentTitle, PageHeader, Spinner } from 'components/common';
+import withParams from 'routing/withParams';
+import withLocation from 'routing/withLocation';
+import StreamsStore from 'stores/streams/StreamsStore';
+import useCurrentUser from 'hooks/useCurrentUser';
+
+const StreamEditPage = ({ params, location }) => {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/pages/StreamEditPage.tsx:28]: Can you please type the props here?

File: graylog2-web-interface/src/contexts/CurrentUserPreferencesProvider.test.tsx [hunk 3]
```diff
@@ -52,6 +53,10 @@ describe('CurrentUserPreferencesProvider', () => {
     return consume;
   };
 
+  beforeEach(() => {
+    asMock(useCurrentUser).mockReturnValue(adminUser);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/contexts/CurrentUserPreferencesProvider.test.tsx:57]: I think we can remove this and leave it to the default mock.

File: graylog2-web-interface/src/pages/EditEventDefinitionPage.test.tsx [hunk 4]
```diff
@@ -64,35 +64,29 @@ jest.mock('stores/event-definitions/EventDefinitionsStore', () => ({
...
-    return (
-      <CurrentUserContext.Provider value={currentUser}>
-        {EditEventDefinitionPageWithParams}
-      </CurrentUserContext.Provider>
-    );
-  };
-
-  EditEventDefinitionPageWithPermissions.defaultProps = { permissions: [] };
+  beforeEach(() => {
+    asMock(useCurrentUser).mockReturnValue(adminUser);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/pages/EditEventDefinitionPage.test.tsx:72]: I think we can remove this and leave it to the default mock.

File: graylog2-web-interface/src/pages/ViewEventDefinitionPage.test.tsx [hunk 5]
```diff
@@ -51,35 +54,32 @@ jest.mock('stores/event-notifications/EventNotificationsStore', () => ({
...
-    return (
-      <CurrentUserContext.Provider value={user}>
-        {EventDefintionPageWithParams}
-      </CurrentUserContext.Provider>
-    );
-  };
-
-  EventDefinitionPageWithPermissions.defaultProps = { permissions: [] };
+  beforeEach(() => {
+    asMock(useCurrentUser).mockReturnValue(adminUser);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/pages/ViewEventDefinitionPage.test.tsx:58]: I think we can remove this and leave it to the default mock.

File: graylog2-web-interface/src/views/components/contexts/SearchPagePreferencesProvider.test.tsx [hunk 6]
```diff
@@ -48,19 +47,28 @@ jest.mock('logic/local-storage/Store', () => ({
...
+jest.mock('views/stores/ViewStore', () => ({
+  ViewStore: {
+    listen: jest.fn(() => () => {}),
+    getInitialState: jest.fn(() => ({})),
+  },
+}));
+
 describe('SearchPagePreferencesProvider', () => {
+  beforeEach(() => {
+    asMock(useCurrentUser).mockReturnValue(adminUser);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/contexts/SearchPagePreferencesProvider.test.tsx:59]: I think we can remove this and leave it to the default mock.

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good! Just a couple of minor suggestions inline.

**Comment by linuspahl** [comment]
@dennisoelkers Regarding the usage of `asMock(useCurrentUser).mockReturnValue(adminUser);` in `beforeEach`, I am doing this whenever we are mocking the return value in one or more test cases, to ensure we reset the return value for all following test cases.

**Comment by dennisoelkers** [comment]
@linuspahl: I understand the reasoning, but I think it is not needed in at least a couple of test suites, where we can remove them for now. The benefit is that we have a single mock implementation which we can maintain centrally.

**Review by dennisoelkers — APPROVED** [review]

**Pull request merged**